### DISCUSSION TITLE
querying whole phrases, and not just the words in the phrase

### DISCUSSION THREAD

**Discussion by designbymichael** [krisk/Fuse#739] [Q&A]
Question for the more experienced Fuse users/developers.  Like most, I struggle with tuning the search results, and it would be so much more reliable if there was a way to search either full phrases as a whole unit (e.g. "dark side of the moon", only results with "dark side of the moon" in the content/keywords matches, as well as any word in the phrase (which is what it's doing now for me, searching "dark side of the moon" would return pages with the word "dark" or "side" or "of" or "the" or "moon").

In the sample code that I've been using [this sample code](https://makewithhugo.com/add-search-to-a-hugo-site/) to implement Fuse, and within it all is the line
`var result = fuse.search(searchQuery);`
and it seems to be just rudimentary search with no sort of options for refining the search query.

I've tried things like:
`var result = fuse.search("="+searchQuery);`
to provide an exact match to the phrase as well as in the search term itself (searching with `=dark side of the moon` and `'dark'side'of'the'moon` , but that didn't seem to produce any relevant results.

What am I missing to take a phrase that someone searches with, and to return just pages that contain that phrase as-is (and not some combination of all the words scattered throughout the page)?  I'd love to be able to outweigh pages with a combination of those words with pages that have the entire phrase present.

**Comment by krisk** [comment]
This is addressed in the `feat/token-search` branch. The new `useTokenSearch` option splits multi-word queries into individual terms, fuzzy-matches each independently, and ranks results by relevance using IDF weighting. So searching "dark side of the moon" will prioritize pages containing more of those words, with exact matches ranking highest.

```js
const fuse = new Fuse(docs, {
  useTokenSearch: true,
  keys: ['content', 'keywords']
})
fuse.search('dark side of the moon')
```
